rpcclient/client.go

```diff
@@ -1,13 +1,85 @@
 package rpcclient
 
-// Client represents a swap RPC client, used to interact with a swap daemon via JSON-RPC calls.
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/rpc/v2/json2"
+)
+
+var (
+	contentTypeJSON   = "application/json"
+	dialTimeout       = 60 * time.Second
+	httpClientTimeout = 30 * time.Minute
+	callTimeout       = 30 * time.Minute
+
+	transport = &http.Transport{
+		DialContext: (&net.Dialer{
+			Timeout: dialTimeout,
+		}).DialContext,
+	}
+	httpClient = &http.Client{
+		Transport: transport,
+		Timeout:   httpClientTimeout,
+	}
+)
+
+// Client primarily exists to be a JSON-RPC client to swapd instances, but it can be used
+// to POST JSON-RPC requests to any JSON-RPC server. Its current use case assumes swapd is
+// running on the local host of a single use system. TLS and authentication are not
+// currently supported.
 type Client struct {
+	ctx      context.Context
 	endpoint string
 }
 
-// NewClient ...
-func NewClient(endpoint string) *Client {
+// NewClient creates a new JSON-RPC client for the specified endpoint. The passed context
+// is used for the full lifetime of the client.
+func NewClient(ctx context.Context, endpoint string) *Client {
 	return &Client{
+		ctx:      ctx,
 		endpoint: endpoint,
 	}
 }
+
+// Post makes a JSON-RPC call to the client's endpoint, serialising any passed request
+// object and deserializing any passed response object from the POST response body. Nil
```

Comment:
That is so weird. The linter forced me to rewrite "serializing" with an "s", but it is allowing "deserializing".  I will harmoniZe the spelling of these two words after the linter update makes it to master.  :)

Reply:
that's painful lol